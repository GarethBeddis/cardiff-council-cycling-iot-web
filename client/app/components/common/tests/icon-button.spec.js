import React from 'react'
import { shallow } from 'enzyme'

import IconButton from '../icon-button'

// Prepare a mock function and mock image component
const mockOnClick = jest.fn()
const mockImg = () => {
  return (
    <svg />
  )
}

describe('Icon Button Component', () => {
  // Reset the Mock
  beforeEach(() => {
    mockOnClick.mockClear()
  })
  // General Tests
  it('Renders without crashing', () => {
    shallow(<IconButton icon={mockImg} />)
  })
  it('OnClick event fires', () => {
    let wrapper = shallow(<IconButton onClick={mockOnClick} icon={mockImg} />)
    wrapper.find('a.icon-btn').first().simulate('click')
    expect(mockOnClick).toHaveBeenCalledTimes(1)
  })
  // Snapshot Tests
  describe('Snapshots', () => {
    it('Standard icon button style matches Snapshot', () => {
      let wrapper = shallow(<IconButton icon={mockImg} title='Test' />)
      expect(wrapper).toMatchSnapshot()
    })
    it('Custom class icon button style matches Snapshot', () => {
      let wrapper = shallow(<IconButton className='test' icon={mockImg} title='Test' />)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
