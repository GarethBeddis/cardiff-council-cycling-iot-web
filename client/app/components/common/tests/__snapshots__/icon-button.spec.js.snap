// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Button Component Snapshots Custom class icon button style matches Snapshot 1`] = `
<a
  className="icon-btn test"
  title="Test"
>
  <div
    className="hover-circle"
  />
  <mockImg
    className="icon"
  />
</a>
`;

exports[`Icon Button Component Snapshots Standard icon button style matches Snapshot 1`] = `
<a
  className="icon-btn"
  title="Test"
>
  <div
    className="hover-circle"
  />
  <mockImg
    className="icon"
  />
</a>
`;
